test/e2e/framework/assertions.go

```diff
@@ -0,0 +1,198 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/pkg/errors"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (f *Framework) AssertDeploymentExistsAndRollout(objName, namespace string) func(*testing.T) {
+	return func(t *testing.T) {
+		err := f.OperatorClient.WaitForDeploymentRollout(context.TODO(), &appsv1.Deployment{
```

Comment:
we shouldn't pass a `context.TODO()`, we could set it to `context.Background()` like the other assertions